File: VB/Scaffolding.DetailCollections/DepartmentContextDataModel/IDepartmentContextUnitOfWork.vb
﻿Imports System
Imports System.Linq
Imports System.Data
Imports System.Data.Entity
Imports System.Linq.Expressions
Imports System.Collections.Generic
Imports Scaffolding.DetailCollections.Common.Utils
Imports Scaffolding.DetailCollections.Common.DataModel
Imports Scaffolding.DetailCollections.Common.DataModel.EntityFramework
Imports Scaffolding.DetailCollections.Model

Namespace Scaffolding.DetailCollections.DepartmentContextDataModel
    ''' <summary>
    ''' IDepartmentContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    ''' </summary>
    Public Interface IDepartmentContextUnitOfWork
        Inherits IUnitOfWork

        ''' <summary>
        ''' The Course entities repository.
        ''' </summary>
        ReadOnly Property Courses() As IRepository(Of Course, Integer)

        ''' <summary>
        ''' The Department entities repository.
        ''' </summary>
        ReadOnly Property Departments() As IRepository(Of Department, Integer)

        ''' <summary>
        ''' The Employee entities repository.
        ''' </summary>
        ReadOnly Property Employees() As IRepository(Of Employee, Integer)
    End Interface
End Namespace

File: CS/Scaffolding.DetailCollections/DepartmentContextDataModel/IDepartmentContextUnitOfWork.cs
using System;
using System.Linq;
using System.Data;
using System.Linq.Expressions;
using System.Collections.Generic;
using Scaffolding.DetailCollections.Common.Utils;
using Scaffolding.DetailCollections.Common.DataModel;
using Scaffolding.DetailCollections.Common.DataModel.DesignTime;
using Scaffolding.DetailCollections.Common.DataModel.EntityFramework;
using Scaffolding.DetailCollections.Model;

namespace Scaffolding.DetailCollections.DepartmentContextDataModel {
    /// <summary>
    /// IDepartmentContextUnitOfWork extends the IUnitOfWork interface with repositories representing specific entities.
    /// </summary>
    public interface IDepartmentContextUnitOfWork : IUnitOfWork {

        /// <summary>
        /// The Course entities repository.
        /// </summary>
        IRepository<Course, int> Courses { get; }

        /// <summary>
        /// The Department entities repository.
        /// </summary>
        IRepository<Department, int> Departments { get; }

        /// <summary>
        /// The Employee entities repository.
        /// </summary>
        IRepository<Employee, int> Employees { get; }
    }
}
